lamindb/models/schema.py

```diff
@@ -626,11 +627,27 @@ def registry(self) -> str:
     def registry(self, value) -> None:
         self.itype = value
 
+    @property
+    def slots(self) -> dict[str, Schema]:
+        """Slots."""
+        if hasattr(self, "_slots"):
+            return self._slots
+        if self.itype == "Composite":
+            self._slots = {
+                link.slot: link.component
+                for link in self.components.through.filter(composite_id=self.id).all()
```

Comment:
This implementation is ultra-slow because it makes one db-request for `link.component`.

It'd be better to make a bulk-request that fetches it at all in one join.